skill_scanner/core/static_analysis/dataflow/forward_analysis.py

```
@@ -498,34 +502,42 @@ def _expr_uses_var(self, expr: ast.AST, var_name: str, fact: ForwardFlowFact) ->
-                # Check transitive dependencies with source sensitivity
-                node_shape = fact.shape_env.get(node.id)
-                node_taint = node_shape.get_taint()
+                if target_labels and node_taint.labels & target_labels:
+                    return True
 
-                if node_taint.is_tainted():
-                    if expected_label in node_taint.labels:
-                        return True
+                # Check structural shapes
+                if node_shape.is_object:
+                    for field_name, field_shape in node_shape.fields.items():
```

Ruff (0.15.18)

[warning] Loop control variable `field_name` not used within loop body

Rename unused `field_name` to `_field_name`

(B007)